### PR TITLE
Refactor IOUCurrencySelection class to use hooks

### PR DESCRIPTION
**Pull request by Julesssss**
CC @marcaaron 

### Details
Simply refactors the IOUCurrencySelection file away from class and introduces hooks.

### Fixed Issues

$ https://github.com/Expensify/App/issues/16274

### Tests

**Request Money**
- From the FAB menu, request money from any user
- At the amount step, tap the currency
- You should be able to select a different currency by searching or scrolling
- At the final step of the flow you should be able to edit the currency
- The request should succeed when created
- [x] Verify that no errors appear in the JS console

**Split Bill**
- From within a group chat, split bill 
- At the amount step, tap the currency
- You should be able to select a different currency by searching or scrolling
- At the final step of the flow you should be able to edit the currency
- The split and requests should succeed when created
- [ ] Verify that no errors appear in the JS console

**Send Money**
- _Only testable if you have Expensify wallets configured, (C+ can ignore)_
- From the FAB menu, send money to any user
- At the amount step, tap the currency
- Search for a currency to filter results
- Delete the search results, the filtering should be removed
- At the final step of the flow you should be able to edit the currency
- The request should succeed when created
- [ ] Verify that no errors appear in the JS console

### Offline tests

**Request Money**
- From the FAB menu, request money from any user
- At the amount step, tap the currency
- You should be able to select a different currency by searching or scrolling
- At the final step of the flow you should be able to edit the currency
- The request should succeed when created
- [ ] Verify that no errors appear in the JS console

**Split Bill**
- From within a group chat, split bill 
- At the amount step, tap the currency
- You should be able to select a different currency by searching or scrolling
- At the final step of the flow you should be able to edit the currency
- The split and requests should succeed when created
- [ ] Verify that no errors appear in the JS console

**Send Money**
- _Only testable if you have Expensify wallets configured, (C+ can ignore)_
- From the FAB menu, send money to any user
- At the amount step, tap the currency
- Search for a currency to filter results
- Delete the search results, the filtering should be removed
- At the final step of the flow you should be able to edit the currency
- The request should succeed when created
- [ ] Verify that no errors appear in the JS console

### QA Steps

**Request Money**
- From the FAB menu, request money from any user
- At the amount step, tap the currency
- You should be able to select a different currency by searching or scrolling
- At the final step of the flow you should be able to edit the currency
- The request should succeed when created
- [ ] Verify that no errors appear in the JS console

**Split Bill**
- From within a group chat, split bill 
- At the amount step, tap the currency
- You should be able to select a different currency by searching or scrolling
- At the final step of the flow you should be able to edit the currency
- The split and requests should succeed when created
- [ ] Verify that no errors appear in the JS console

**Send Money**
- _Only testable if you have Expensify wallets configured, (C+ can ignore)_
- From the FAB menu, send money to any user
- At the amount step, tap the currency
- Search for a currency to filter results
- Delete the search results, the filtering should be removed
- At the final step of the flow you should be able to edit the currency
- The request should succeed when created
- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Screenshot 2023-06-12 at 16 21 48

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Simulator Screeqqqn Shot - iPhone 14 - 2023-06-12 at 16 24 34

</details>

<details>
<summary>Mobile Web - Safari</summary>

Simulator Screen Shot - iPhone 14 - 2023-06-12 at 16 24 34

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-06-12 at 16 22 30

</details>

<details>
<summary>iOS</summary>

Simulator Screen Shot - iPhone 14 - 2023-06-12 at 16 26 35

</details>

<details>
<summary>Android</summary>

Screenshot_20230613_141900

</details>

### CODE REVIEW

**Comment by Julesssss** [comment]
Hey both, I'm struggling to figure out why [this](https://github.com/Expensify/App/pull/20579/files#diff-496700da68574808c5981ac4884e39d73ea02df4aa59bd5869e0c9227defe4c2R107-R108) lint change is necessary.

File: src/pages/iou/IOUCurrencySelection.js [hunk 1]
```diff
@@ -56,125 +56,98 @@ const defaultProps = {
...
-                                'keyForList',
-                            )}
-                            shouldHaveOptionSeparator
-                        />
-                    </>
-                )}
-            </ScreenWrapper>
-        );
-    }
+    }, [searchValue, currencyData, props.translate]);
```
Comment (Julesssss) [inline src/pages/iou/IOUCurrencySelection.js:107]: Surely it's better to just pass `props.translate` rather than `props`?

Comment (narefyev91) [reply src/pages/iou/IOUCurrencySelection.js:107]: for sure - props.translate should be in deps

File: src/pages/iou/IOUCurrencySelection.js [hunk 2]
```diff
@@ -56,125 +56,98 @@ const defaultProps = {
...
+            if (_.isEmpty(backTo) || props.navigation.getState().routes.length === 1) {
+                Navigation.goBack();
+            } else {
+                Navigation.navigate(`${props.route.params.backTo}?currency=${option.currencyCode}`);
+            }
+        },
+        [props.route, props.navigation],
+    );
+
+    const [searchValue, setCurrentSearchValue] = useState('');
```
Comment (narefyev91) [inline src/pages/iou/IOUCurrencySelection.js:91]: let's clean up a little bit - useState's logic should be define on the top of the function

File: src/pages/iou/IOUCurrencySelection.js [hunk 3]
```diff
@@ -56,125 +56,98 @@ const defaultProps = {
...
+    const getSections = useMemo(() => {
+        if (searchValue.trim() && !currencyData.length) {
             return [];
         }
         const sections = [];
         sections.push({
-            title: this.props.translate('iOUCurrencySelection.allCurrencies'),
-            data: this.state.currencyData,
+            title: props.translate('iOUCurrencySelection.allCurrencies'),
+            data: currencyData,
```
Comment (narefyev91) [inline src/pages/iou/IOUCurrencySelection.js:101]: i think you do not need sections as variable
Just simply return on line 106 
```
return [{title: props.translate('iOUCurrencySelection.allCurrencies'),
            data: currencyData,            
            shouldShow: true,
            indexOffset: 0,}]
```

File: src/pages/iou/IOUCurrencySelection.js (hunk 1 above)
Comment (Julesssss) [reply src/pages/iou/IOUCurrencySelection.js:107]: Thanks. I've read up on prop destructuring, is this how it's supposed to be done? 

`const { translate } = props;`

Comment (Julesssss) [reply src/pages/iou/IOUCurrencySelection.js]: I think I've done this correctly: https://github.com/Expensify/App/pull/20579/files#diff-496700da68574808c5981ac4884e39d73ea02df4aa59bd5869e0c9227defe4c2R94

**Comment by Julesssss** [comment]
>  melvin-bot bot requested review from allroundexperts and AndrewGable and removed request for Expensify/pullerbear 2 hours ago

Not sure why this happened, but please ignore @AndrewGable @allroundexperts

**Comment by Julesssss** [comment]
Okay, I fixed an issue where deleting the query wouldn't update the filtering, applied the suggested changes, and I think it's ready for review.

File: src/pages/iou/IOUCurrencySelection.js [hunk 4]
```diff
@@ -56,125 +56,99 @@ const defaultProps = {
...
+                    <HeaderWithBackButton
+                        title={props.translate('iOUCurrencySelection.selectCurrency')}
+                        onBackButtonPress={() => Navigation.goBack(ROUTES.getIouRequestRoute(Navigation.getTopmostReportId()))}
+                    />
+                    <OptionsSelector
+                        sections={getSections}
+                        onSelectRow={confirmCurrencySelection}
+                        value={searchValue}
+                        onChangeText={changeSearchValue}
+                        textInputLabel={props.translate('common.search')}
```
Comment (narefyev91) [inline src/pages/iou/IOUCurrencySelection.js:136]: same here

File: src/pages/iou/IOUCurrencySelection.js [hunk 5]
```diff
@@ -56,125 +56,99 @@ const defaultProps = {
...
+            const filteredCurrencies = _.filter(currencyOptions, (currencyOption) => searchRegex.test(currencyOption.text));
+
+            setCurrentSearchValue(searchQuery);
+            setCurrencyData(filteredCurrencies);
+        },
+        [currencyOptions],
+    );
+
+    const headerMessage = searchValue.trim() && !currencyData.length ? props.translate('common.noResultsFound') : '';
+
```
Comment (narefyev91) [inline src/pages/iou/IOUCurrencySelection.js:122]: and here

**Comment by Julesssss** [comment]
Updated!

**Comment by Julesssss** [comment]
FYI I'm ooo tomorrow but will catch up with any comments on Thursday. Unless I get lucky and no further changes are necessary 🤞 

@youssef-lr @fedirjh @narefyev91

**Review by fedirjh — COMMENTED** [review]
@Julesssss I have an extra suggestion to simplify and optimise the logic of this component.

File: src/pages/iou/IOUCurrencySelection.js [hunk 6]
```diff
@@ -56,125 +56,99 @@ const defaultProps = {
...
+                    currencyCode,
+                    keyForList: currencyCode,
+                    customIcon: isSelectedCurrency ? greenCheckmark : undefined,
+                    boldStyle: isSelectedCurrency,
+                };
+            }),
+        [selectedCurrencyCode, props.currencyList],
+    );
+
+    const [currencyData, setCurrencyData] = useState(currencyOptions);
```
[the remark this answers is not in the document]
Comment (narefyev91) [reply src/pages/iou/IOUCurrencySelection.js:77]: getSections should be renamed to sections - it's returning array, 'get' mostly adds to functions

[the remark this answers is not in the document]
Comment (Julesssss) [reply src/pages/iou/IOUCurrencySelection.js:77]: Ah thanks, I see now.

**Review by Julesssss — COMMENTED** [review]
Thanks both. So if I understand correctly there was no need to split up the individual states, as they are all tied together and get updated at the same time?

**Comment by Julesssss** [comment]
I think I have successfully merged the updates to a single `useMemo`. Please review again

File: src/pages/iou/IOUCurrencySelection.js [hunk 7]
```diff
@@ -108,73 +87,56 @@ class IOUCurrencySelection extends Component {
...
+                      {
+                          title: translate('iOUCurrencySelection.allCurrencies'),
+                          data: filteredCurrencies,
+                          shouldShow: true,
+                          indexOffset: 0,
+                      },
+                  ],
+            headerMessage: isEmpty ? translate('common.noResultsFound') : '',
+        };
+    }, [props.currencyList, searchValue, selectedCurrencyCode, translate]);
```
Comment (narefyev91) [inline src/pages/iou/IOUCurrencySelection.js:112]: maybe we can also add currencyList at line 78?

**Comment by narefyev91** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Review by fedirjh — APPROVED** [review]
Nice work , LGTM 🚀

**Comment by narefyev91** [comment]
## Reviewer Checklist
- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>


https://github.com/Expensify/App/assets/28352309/996845ad-0305-4a12-a70d-77fb2869567c


https://github.com/Expensify/App/assets/28352309/8b51574e-737e-4160-b863-529843a76c95




</details>

<details>
<summary>Mobile Web - Chrome</summary>


https://github.com/Expensify/App/assets/28352309/4c0e7946-fa30-455b-80f1-35c875d418cb


https://github.com/Expensify/App/assets/28352309/7b19c606-f76a-4a4d-b319-5ad76be38d3c




</details>

<details>
<summary>Mobile Web - Safari</summary>


https://github.com/Expensify/App/assets/28352309/c4f1a5d9-d213-4170-91f5-2b566da399bf


https://github.com/Expensify/App/assets/28352309/0da160cc-3bab-4f28-a542-6a7d1fec8402




</details>

<details>
<summary>Desktop</summary>


https://github.com/Expensify/App/assets/28352309/1ea6ff17-50b6-44b3-9a05-8ef20b350f26


https://github.com/Expensify/App/assets/28352309/4c3b46e8-b67a-4ba1-9e4c-ab536fecbb5b




</details>

<details>
<summary>iOS</summary>


https://github.com/Expensify/App/assets/28352309/0a3c93ad-bc97-4fea-907e-76e6c91cd384


https://github.com/Expensify/App/assets/28352309/f77b5307-5158-4cd4-9a6c-ff5401607c25




</details>

<details>
<summary>Android</summary>


https://github.com/Expensify/App/assets/28352309/8a49e3cb-2127-498f-8e33-c07c5987f28e


https://github.com/Expensify/App/assets/28352309/41481742-d279-4053-86ef-28864b4bbb57




</details>

**Review by narefyev91 — APPROVED** [review]
Nice work! LGTM!

**Comment by Julesssss** [comment]
Bumping for final review @youssef-lr

**Comment by Julesssss** [comment]
Hi @fedirjh, I forgot about this one and had to resolve a conflict. Would you mind one final review of the changes since you last approved please?

**Comment by Julesssss** [comment]
@youssef-lr is OOO, so I will reassign once this is ready for internal review.

**Review by fedirjh — APPROVED** [review]
LGTM and tests well.

cc @Julesssss

**Comment by Julesssss** [comment]
Hey @srikarparsi, pullerbear is broken, but you won the reviewer lottery. Assigning you for the code review, C+ has already approved

**Review by srikarparsi — APPROVED** [review]
code looks good to me!

**Comment by dangrous** [comment]
Haven't confirmed yet but it's looking like this caused https://github.com/Expensify/App/issues/22130, I'll look into it and see if I can figure out a fix but wanted to ping early.

File: src/pages/iou/IOUCurrencySelection.js [hunk 8]
```diff
@@ -105,75 +84,56 @@ class IOUCurrencySelection extends Component {
...
+        };
+    }, [currencyList, searchValue, selectedCurrencyCode, translate]);
+
+    return (
+        <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+            {({safeAreaPaddingBottomStyle}) => (
+                <>
+                    <HeaderWithBackButton
+                        title={translate('iOUCurrencySelection.selectCurrency')}
+                        onBackButtonPress={() => Navigation.goBack(ROUTES.getIouRequestRoute(Navigation.getTopmostReportId()))}
```
Comment (dangrous) [inline src/pages/iou/IOUCurrencySelection.js:117]: `ROUTES.getIouRequestRoute()` doesn't currently exist which is causing the regression I commented on, will see if I can figure out what this is supposed to be and add it.

Comment (dangrous) [reply src/pages/iou/IOUCurrencySelection.js:117]: Yep! That's what I ended up going with - https://github.com/Expensify/App/pull/22135/files#diff-496700da68574808c5981ac4884e39d73ea02df4aa59bd5869e0c9227defe4c2R119 - it's weird, I looked through every commit in this PR and it never actually changed away from that in any of them. But it ends up being different in the end. I'm not sure how that happened

Comment (dangrous) [reply src/pages/iou/IOUCurrencySelection.js:117]: ahh that makes sense.

Comment (Julesssss) [reply src/pages/iou/IOUCurrencySelection.js:117]: Thanks both, Yeah my first guess was the merge as I recall some conflicts. I must have missed one when resolving.

**Pull request merged**